processing/src/main/java/org/apache/druid/query/groupby/epinephelinae/vector/VectorGroupByEngine.java

```diff
@@ -76,10 +76,18 @@ public static boolean canVectorize(
     Function<String, ColumnCapabilities> capabilitiesFunction = name ->
         query.getVirtualColumns().getColumnCapabilitiesWithFallback(adapter, name);
 
+    final boolean adapterCanVectorize = adapter.canVectorize(filter, query.getVirtualColumns(), false);
+    final boolean virtualColumnsCanVectorize;
+    if (query.getVirtualColumns().getVirtualColumns().length > 0) {
```

Comment:
There's some duplication here between the query engines — how about creating a helper that takes the query and the virtual columns and creates a `Vectorize` instance that synthesizes both?

Reply:
I'll see if I can clean this up a bit

Reply:
moved into a static method `VirtualColumns.shouldVectorize` which I'm not sure is the best place or not, but at least consolidates the logic